packages/expo-cli/src/commands/upgrade.ts

```diff
@@ -147,6 +147,23 @@ async function upgradeAsync(requestedSdkVersion: string | null, options: Options
     }
   }
 
+  // Can't upgrade if Expo is running
+  let status = await Project.currentStatus(projectRoot);
+  if (status === 'running') {
+    let answer = await prompt({
+      type: 'confirm',
+      name: 'closeExpo',
+      message: 'Expo needs to close before we can upgrade your dependencies. Do you want to close it?',
+    });
+
+    if (!answer.closeExpo) {
+      return;
```

Comment:
In this case the exit status of the command should be non-zero. You can, for example, throw a `CommandError` to achieve this.

Reply:
Thanks! Will do in about 3-4 hours!

Reply:
I added it, not sure if it has the proper name though, maybe `BAD_CHOICE` is also a good one? I didn't add it to the [error codes](https://github.com/expo/expo-cli/blob/master/packages/expo-cli/src/CommandError.ts#L5) because the other two command errors are not listed there. But I'd be happy to add it if necessary.

Reply:
The error code doesn't matter all that much in this case – it's mostly useful for cases where we want to catch a specific error, e.g. a common pattern is like this:
```
try {
  doStuffThatMightFail();
} catch (error) {
  if (error.code === 'BAD_CHOICE') {
    trySomethingElse();
  } else {
    throw error;
  }
}
```
Tl;dr: `BAD_CHOICE` is a good choice! 😆

Reply:
Thanks Ville! ❤️  I'll remember that next time 😄 

For now, I think this specific instance is superseded by Brent's suggestion to just tell instead of ask, right?